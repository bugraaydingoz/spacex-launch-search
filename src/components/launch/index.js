export { Launch } from './Launch'
export { LaunchList } from './LaunchList'
export { LaunchLoading } from './LaunchLoading'
