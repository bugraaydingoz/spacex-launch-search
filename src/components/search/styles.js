import styled from 'styled-components'

export const Input = styled.input`
  padding: 10px 20px;
  margin-bottom: 30px;
  width: 100%;
  background: rgba(0, 0, 0, 0);
  font-size: 20px;
  border: var(--border);
  color: #fff;
`
