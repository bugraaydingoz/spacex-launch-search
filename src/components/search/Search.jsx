import React from 'react'

// Styles
import { Input } from './styles'

export function Search({ value, onChange }) {
  return (
    <Input
      type="text"
      placeholder="Search a launch"
      value={value}
      onChange={(e) => onChange(e.target.value)}
    />
  )
}
