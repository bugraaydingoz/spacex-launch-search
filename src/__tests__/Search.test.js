import React from 'react'
import renderer from 'react-test-renderer'
import { Search } from '../components/search'

test('renders correctly', () => {
  const tree = renderer.create(<Search onChange={() => {}} />).toJSON()
  expect(tree).toMatchSnapshot()
})
