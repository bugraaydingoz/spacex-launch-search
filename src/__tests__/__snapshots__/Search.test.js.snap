// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="sc-AxjAm hxBlYH"
  onChange={[Function]}
  placeholder="Search a launch"
  type="text"
/>
`;
